src/pages/Fox/components/FoxFarming.tsx

```diff
@@ -0,0 +1,365 @@
+
+    return true
+  }, [assetAccountId, appDispatch, foxEthMarketData, opportunityId])
+
+  useEffect(() => {
+    queryClient.invalidateQueries({
+      queryKey: ['fetchOpportunityData'],
+    })
+  }, [assetAccountId, queryClient])
+
+  const { isLoading: isOpportunityQueryLoading, isFetching: isOpportunityFetching } = useQuery({
+    queryKey: ['fetchOpportunityData', assetAccountId],
+    queryFn: fetchOpportunityData,
+    enabled: Boolean(foxEthMarketData),
+  })
+
+  const isOpportunityLoading = useMemo(
+    () => isOpportunityQueryLoading || isOpportunityFetching,
+    [isOpportunityQueryLoading, isOpportunityFetching],
+  )
+
+  const underlyingAsset = useAppSelector(state =>
+    selectAssetById(state, opportunity?.underlyingAssetId ?? ''),
+  )
+
+  const rewardAsset = useAppSelector(state =>
+    selectAssetById(state, opportunity?.rewardAssetIds[0] ?? ''),
+  )
+
+  const handleClick = useCallback(
+    (opportunity: LpEarnOpportunityType, action: DefiAction) => {
+      const {
+        type,
+        provider,
+        contractAddress,
+        chainId,
+        rewardAddress,
+        assetId,
+        highestBalanceAccountAddress,
+      } = opportunity
+      const { assetReference, assetNamespace } = fromAssetId(assetId)
+
+      if (!isConnected || isDemoWallet) {
+        dispatch({ type: WalletActions.SET_WALLET_MODAL, payload: true })
+        return
+      }
+
+      history.push({
+        pathname: location.pathname,
+        search: qs.stringify({
+          type,
+          provider,
+          chainId,
+          contractAddress,
+          assetNamespace,
+          assetReference,
+          highestBalanceAccountAddress,
+          rewardId: rewardAddress,
+          modal: action,
+        }),
+        state: { background: location },
+      })
+    },
+    [dispatch, history, isConnected, isDemoWallet, location],
+  )
+
+  const handleManageClick = useCallback(() => {
+    if (!opportunity) return
+
+    handleClick(opportunity, DefiAction.Overview)
+  }, [opportunity, handleClick])
+
+  const handleClaimClick = useCallback(() => {
+    if (!opportunity) return
+
+    handleClick(opportunity, DefiAction.Claim)
+  }, [opportunity, handleClick])
+
+  const rewardsCryptoAmount = useMemo(() => {
+    if (!opportunity) return
+    if (!rewardAsset) return
+
+    return fromBaseUnit(opportunity.rewardsCryptoBaseUnit?.amounts[0], rewardAsset?.precision ?? 0)
+  }, [opportunity, rewardAsset])
+
+  const totalStakingValue = useMemo(() => {
+    if (!opportunity) return
+
+    return fromBaseUnit(opportunity?.stakedAmountCryptoBaseUnit, underlyingAsset?.precision ?? 0)
+  }, [opportunity, underlyingAsset?.precision])
+
+  const apy = useMemo(() => {
+    if (!opportunity) return
+
+    return opportunity.apy
+  }, [opportunity])
+
+  const nextEpochHuman = useMemo(() => {
+    if (!periodFinish) return
+
+    return formatSecondsToDuration(
+      bnOrZero(periodFinish.toString())
+        .minus(Date.now() / 1000)
```

Comment:
nitpick: could do `dayjs(periodFinish).fromNow()` not to roll out own date calcs

Reply:
Ok but I have to multiply periodFinish by 1000 as it's using seconds and dayjs is using ms

Done in [527f86c](https://github.com/shapeshift/web/pull/7934/commits/527f86ce55071bc81f46808931a053d3e5ee04e3)